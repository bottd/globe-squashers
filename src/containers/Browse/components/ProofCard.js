import React from 'react';

export default function ProofCard({url, title, note}) {
  return (
    <div className="ProofCard">
      <h3>{title}</h3>
      <img src={url} alt={title} />
      <p>{note}</p>
    </div>
  );
}
