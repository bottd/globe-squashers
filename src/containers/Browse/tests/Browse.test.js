import React from 'react';
import {shallow} from 'enzyme';
import {Browse, mapStateToProps, mapDispatchToProps} from '../';
import ProofCard from '../components/ProofCard';

describe('Browse', () => {
  describe('Browse component', () => {
    it('should render like snapshot', () => {
      const wrapper = shallow(<Browse proofs={[]} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('ProofCard component', () => {
    it('Should render like snapshot', () => {
      const mockProof = {
        url: 'image.com',
        title: 'A title',
        note: 'This is fake'
      }
      const wrapper = shallow(<ProofCard {...mockProof}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Map state to props', () => {
    it('Should return the proofs from state', () => {
      const mockState = {
        proofs: [],
        other: [],
      };
      const expected = {
        proofs: [],
      };
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });
  });

  describe('Map dispatch to props', () => {
    it('Should return an empty object', () => {
      const result = mapDispatchToProps(jest.fn());
      expect(result).toEqual({});
    });
  });
});
