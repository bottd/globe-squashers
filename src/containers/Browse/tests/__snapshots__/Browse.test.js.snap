// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse Browse component should render like snapshot 1`] = `
<div
  className="Browse"
>
  <p>
    No proof here, go to the create tab and start working!
  </p>
</div>
`;

exports[`Browse ProofCard component Should render like snapshot 1`] = `
<div
  className="ProofCard"
>
  <h3>
    A title
  </h3>
  <img
    alt="A title"
    src="image.com"
  />
  <p>
    This is fake
  </p>
</div>
`;
